pkg/test/mocks/client.go

```diff
@@ -48,8 +48,14 @@ func (FakeClient) List(ctx context.Context, list runtime.Object, opts ...client.
 	panic("not implemented")
 }
 
-func (FakeClient) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOptionFunc) error {
-	panic("not implemented")
+func (f FakeClient) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOptionFunc) error {
+	resource := reflect.ValueOf(obj).Elem()
+	namespace := resource.FieldByName("ObjectMeta").FieldByName("Namespace").String()
+	name := resource.FieldByName("ObjectMeta").FieldByName("Name").String()
+
+	key := client.ObjectKey{Namespace: namespace, Name: name}
+	f.c[key] = obj
```

Comment:
Maybe error or panic if it already exists?

Though it's a fine line between mock and re-implementing the apiserver :)